client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -12,10 +20,175 @@ import CardHeading from 'calypso/components/card-heading';
+				'This purchase is currently billed to your PayPal account (%(emailAddress)s).',
+				{
+					args: {
+						emailAddress: storedPaymentAgreement.email,
+					},
+				}
+			);
+			return <Notice { ...noticeProps } />;
+		}
+
+		noticeProps.text = translate( 'This purchase is currently billed to your PayPal account.' );
+		return <Notice { ...noticeProps } />;
+	}
+
+	return null;
+}
+
+function AllSubscriptionsEffectWarning( {
+	useForAllSubscriptions,
+}: {
+	useForAllSubscriptions: boolean;
+} ) {
 	const translate = useTranslate();
 
+	if ( useForAllSubscriptions ) {
+		return (
+			<span className="payment-method-add__all-subscriptions-effect-warning">
+				{ translate( 'This card will be used for future renewals of existing purchases.' ) }
+			</span>
+		);
+	}
+	return (
+		<span className="payment-method-add__all-subscriptions-effect-warning">
+			{ translate(
+				'This card will not be assigned to any subscriptions. You can assign it to a subscription from the subscription page.'
+			) }
+		</span>
+	);
+}
+
+function PaymentMethodAdd(): ReactElement {
+	const purchase = undefined;
+	const reduxDispatch = useDispatch();
+	const translate = useTranslate();
+	const { isStripeLoading, stripeLoadingError, stripeConfiguration, stripe } = useStripe();
+
+	const stripeMethod = useCreateStoredCreditCard( {
+		isStripeLoading,
+		stripeLoadingError,
+		stripeConfiguration,
+		stripe,
+		shouldUseEbanx: false,
+		shouldShowTaxFields: true,
+		activePayButtonText: translate( 'Save card' ),
+	} );
+	const paymentMethods = useMemo( () => [ stripeMethod ].filter( Boolean ), [ stripeMethod ] );
+
+	useEffect( () => {
+		if ( stripeLoadingError ) {
+			reduxDispatch( errorNotice( stripeLoadingError.message ) );
+		}
+	}, [ stripeLoadingError, reduxDispatch ] );
+
+	const goToPaymentMethods = () => page( '/partner-portal/payment-method/' );
+
+	const currentlyAssignedPaymentMethodId = getPaymentMethodIdFromPayment( purchase?.payment );
+
+	const showSuccessMessage = useCallback(
+		( message ) => {
+			reduxDispatch( successNotice( message, { displayOnNextPage: true, duration: 5000 } ) );
+		},
+		[ reduxDispatch ]
+	);
+
+	const showErrorMessage = useCallback(
+		( error ) => {
+			const message = error?.toString ? error.toString() : error;
+			reduxDispatch( errorNotice( message, { displayOnNextPage: true } ) );
+		},
+		[ reduxDispatch ]
+	);
+
+	const showInfoMessage = useCallback(
+		( message ) => {
+			reduxDispatch( infoNotice( message ) );
+		},
+		[ reduxDispatch ]
+	);
+
+	const currentPaymentMethodNotAvailable = ! paymentMethods.some(
+		( paymentMethod ) => paymentMethod.id === currentlyAssignedPaymentMethodId
+	);
+
+	const [ useForAllSubscriptions, setUseForAllSubscriptions ] = useState< boolean >( ! purchase );
+
+	const assignAllSubscriptionsText = String(
+		translate( 'Assign this payment method to all of my subscriptions' )
+	);
+
+	if ( paymentMethods.length === 0 ) {
+		return <div>{ translate( 'loading...' ) }</div>;
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **45** times:
`translate( 'Loading…', { context: 'Sharing: Publicize status pending button label'} )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-partner-portal-add-payment-method-3)